src/v/kafka/server/tests/member_test.cc

```diff
@@ -175,4 +176,134 @@ SEASTAR_THREAD_TEST_CASE(member_serde) {
+    BOOST_REQUIRE(new_out.members[1].client_host == "");
+}
+
+// new code + new version. will see client id/host
+SEASTAR_THREAD_TEST_CASE(group_log_group_metadata_compat1) {
+    group_log_group_metadata md;
+    md.protocol_type = kafka::protocol_type("asdf");
+    md.state_timestamp = 333333;
+
+    md.members.push_back(member_state{
+      .id = member_id("foobar"),
+      .protocols = {{.name = protocol_name("asdf")}},
+      .client_id = kafka::client_id("c0"),
+      .client_host = kafka::client_host("c1"),
+    });
+    md.members.push_back(member_state{
+      .id = member_id("foobar1"),
+      .protocols = {{.name = protocol_name("asdf1")}},
+      .client_id = kafka::client_id("c2"),
+      .client_host = kafka::client_host("c3"),
+    });
+
+    auto old_buf = reflection::to_iobuf(std::move(md));
+
+    auto new_out = reflection::from_iobuf<group_log_group_metadata>(
+      std::move(old_buf));
+
+    BOOST_REQUIRE(new_out.protocol_type == "asdf");
+    BOOST_REQUIRE(new_out.state_timestamp == 333333);
+    BOOST_REQUIRE(new_out.members.size() == 2);
+    BOOST_REQUIRE(new_out.members[0].id == "foobar");
+    BOOST_REQUIRE(new_out.members[0].protocols.size() == 1);
+    BOOST_REQUIRE(new_out.members[0].protocols[0].name == "asdf");
+    BOOST_REQUIRE(new_out.members[0].client_id == "c0");
+    BOOST_REQUIRE(new_out.members[0].client_host == "c1");
+    BOOST_REQUIRE(new_out.members[1].id == "foobar1");
+    BOOST_REQUIRE(new_out.members[1].protocols.size() == 1);
+    BOOST_REQUIRE(new_out.members[1].protocols[0].name == "asdf1");
+    BOOST_REQUIRE(new_out.members[1].client_id == "c2");
+    BOOST_REQUIRE(new_out.members[1].client_host == "c3");
+}
+
+// old code + new version. doesn't see client id/host
+SEASTAR_THREAD_TEST_CASE(group_log_group_metadata_compat2) {
+    group_log_group_metadata md;
+    md.protocol_type = kafka::protocol_type("asdf");
+    md.state_timestamp = 333333;
+
+    md.members.push_back(member_state{
+      .id = member_id("foobar"),
+      .protocols = {{.name = protocol_name("asdf")}},
+      .client_id = kafka::client_id("c0"),
+      .client_host = kafka::client_host("c1"),
+    });
+    md.members.push_back(member_state{
+      .id = member_id("foobar1"),
+      .protocols = {{.name = protocol_name("asdf1")}},
+      .client_id = kafka::client_id("c2"),
+      .client_host = kafka::client_host("c3"),
+    });
+
+    auto old_buf = reflection::to_iobuf(std::move(md));
+
+    auto new_out = reflection::from_iobuf<group_log_group_metadata_old>(
+      std::move(old_buf));
+
+    BOOST_REQUIRE(new_out.protocol_type == "asdf");
+    BOOST_REQUIRE(new_out.state_timestamp == 333333);
+    BOOST_REQUIRE(new_out.members.size() == 2);
+    BOOST_REQUIRE(new_out.members[0].id == "foobar");
+    BOOST_REQUIRE(new_out.members[0].protocols.size() == 1);
+    BOOST_REQUIRE(new_out.members[0].protocols[0].name == "asdf");
+    // BOOST_REQUIRE(new_out.members[0].client_id == "c0");
```

Comment:
remove comments?

Reply:
ah! those were intention because in that decoding scenario the data is in the source but not in the version of the code simulating the decoding.